webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -70,1045 +73,1346 @@ class SearchResponderV2 extends Responder {
+                                            filterCompare.operator match {
 
-                            case _ => AdditionalStatements()
+                                                case CompareExpressionOperator.EQUALS =>
 
-                        }
+                                                    // any overlap in considered as equality
+                                                    val leftArgFilter = CompareExpression(XsdLiteral(date.dateval1.toString, OntologyConstants.Xsd.Integer), CompareExpressionOperator.LESS_THAN_OR_EQUAL_TO, dateValueHasEndVar)
 
-                    } else {
-                        throw NotImplementedException(s"Extended search not implemented for schema $apiSchema")
-                    }
+                                                    val rightArgFilter = CompareExpression(XsdLiteral(date.dateval2.toString, OntologyConstants.Xsd.Integer), CompareExpressionOperator.GREATER_THAN_OR_EQUAL_TO, dateValueHasStartVar)
 
-                case OntologyConstants.KnoraBase.StillImageFile =>
-                    if (apiSchema == ApiV2Schema.SIMPLE) {
+                                                    val filter = AndExpression(leftArgFilter, rightArgFilter)
 
-                        statementPattern.obj match {
+                                                    TransformedFilterExpression(
+                                                        filter,
+                                                        Seq(
+                                                            dateValStartStatement, dateValEndStatement
+                                                        )
+                                                    )
 
-                            case fileVar: ExtendedSearchVar =>
+                                                case CompareExpressionOperator.NOT_EQUALS =>
 
-                                // the statement's object is a variable
+                                                    // no overlap in considered as inequality
+                                                    val leftArgFilter = CompareExpression(XsdLiteral(date.dateval1.toString, OntologyConstants.Xsd.Integer), CompareExpressionOperator.GREATER_THAN, dateValueHasEndVar)
 
-                                // since all value property statements are eliminated, recreate the statement using the given predicate
+                                                    val rightArgFilter = CompareExpression(XsdLiteral(date.dateval2.toString, OntologyConstants.Xsd.Integer), CompareExpressionOperator.LESS_THAN, dateValueHasStartVar)
 
-                                // TODO: maybe only those value property statements have to be eliminated whose object is a literal
+                                                    val filter = OrExpression(leftArgFilter, rightArgFilter)
 
-                                AdditionalStatements(additionalStatements = Vector(
-                      
```

Comment:
It has just occurred to me that this is not good enough since it does not work with property inheritance. We should add alls subproperties of the given property too.

@benjamingeer We have that information in OntologyResponder. We could ask the ontology responder to return all subproperties for a given property.

I think we have to do the same for Filters, where property variables are restricted. We would have to add Or expressions for the subproperties.